src/lib/fetch-document-list.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-console */
+const { info } = require('../lib/log');
+const fs = require('../lib/sync-fs');
+const DOC_TYPES = ['district_hospital', 'health_center', 'clinic', 'person', 'user', 'user-settings', 'contact'];
+
+const jsonDocPath = (directoryPath, docID) => `${directoryPath}/${docID}.doc.json`;
+
+const fetchDocumentListFromDB = async (db, ids) => {
+  info('Downloading doc(s)...');
+  const documentDocs = await db.allDocs({
+    keys: ids,
+    include_docs: true,
+  });
+
+  const errors =  { invalidDocType: [], docNotFound: [] };
+
+  documentDocs.rows.forEach(row => {
+    if (!row.doc) {
+        errors.docNotFound.push(`Document with id '${row.key}' could not be found.`);
+        return;
+    }
+
+    if (!DOC_TYPES.includes(row.doc.type)) {
+      errors.invalidDocType.push(`Document with id ${row.key} of type ${row.doc.type} cannot be edited`);
+    }
+  });
+
+  if (errors.docNotFound.length || errors.invalidDocType.length) {
```

Comment:
There's an unnamed concept in this code, which is: "If there are errors, stop". If we wanted to get all OOP on this, we'd have a method on that errors object to indicate emptiness, but more appropriate for JS is to use [_.isEmpty](https://docs-lodash.com/v4/is-empty/). As the code is currently though, that won't work, since the object is created with 2 properties. Those aren't strictly necessary though, and only seem to be to aid for the collection/push technique. At the moment that side effect is harmless, but if we're to make the `isEmpty` change (please do) it starts to matter and we're better served by having the object truly be empty. That might start to feel fragile since if a future coder makes the "mistake" of putting properties into it, they'll break the `isEmpty` check. That's just a sales pitch for unit testing though :) ...that brings up another good reason to move the error collecting logic out of here: it allows you to unit test error collection logic in isolation and build a stable interface around that function so other logic (like my propsed `isEmpty` check here) can safely rely on it.